frontend/src/app/modules/challenges/challenge-creation/challenge-creation.component.ts

```diff
@@ -84,6 +87,34 @@ export class ChallengeCreationComponent extends BaseComponent implements HasUnsa
         this.steps = this.stepsData.map((s) => s.step);
         this.challenge = getNewChallenge();
         this.challengeVersion = getNewChallengeVersion();
+
+        const data = this.router.getCurrentNavigation()?.extras.state?.['myComplexObject'];
+
+        if (data) {
+            const {
+                category,
+                completeSolution,
+                description,
+                exampleTestCases,
+                initialSolution,
+                level,
+                tags,
+                testCases,
+                title,
+                language,
+            } = data as IChallengeGenerateResponse;
+
+            this.challenge.title = title;
+            this.challenge.instructions = description;
+            this.challenge.category = category;
+            this.challenge.level = level;
+            this.challenge.tags = tags;
+            this.challengeVersion.completeSolution = completeSolution;
+            this.challengeVersion.initialSolution = initialSolution;
+            this.challengeVersion.exampleTestCases = exampleTestCases;
+            this.challengeVersion.testCases = testCases;
+            this.challengeVersion.languageId = language?.id || 0;
+        }
```

Comment:
not in consltucroe
and use spread operator for defending properties

and why || 0? default values should be null or underfined